tools/serve/serve.py

```diff
@@ -543,23 +543,19 @@ def create_daemon(self, init_func, host, port, paths, routes, bind_address,
 
         if self.daemon:
             try:
-                self.daemon.start(block=False)
+                self.daemon.start()
                 try:
-                    self.stop.wait()
+                    self.stop_flag.wait()
```

Comment:
Do we want a timeout here?

Reply:
I don't think that makes sense? The stop flag is an IPC mechanism to signal to the server that it should shut down; the process owning the server waits on the flag whilst running the server itself in another thread. I suppose the failure mode here is "the other thread stopped and we're still waiting", but it would be a larger change to fix that (we'd need to add a method to the server objects to check the serving thread is alive and poll that periodically, or something).